spec/custom/index.html

```diff
@@ -584,167 +457,224 @@
-    <li>Let <var>DEFINITION</var> be an <a>element definition</a> that has <var>PROTOTYPE</var> as <a>custom element prototype</a></li>
-    <li>Let <var>CONSTRUCTOR</var> be the <a href="https://heycam.github.io/webidl/#dfn-interface-object">interface object</a> whose <a href="https://heycam.github.io/webidl/#dfn-interface-prototype-object">interface prototype object</a> is <var>PROTOTYPE</var> and when called as a constructor, executes these steps:
-    <ol>
-        <li>Let <var>ELEMENT</var> be the <a href="https://dom.spec.whatwg.org/#context-object">context object</a></li>
-        <li>Let <var>TYPE</var> be the <a>custom element type</a> in <var>DEFINITION</var></li>
-        <li>Let <var>NAME</var> be the <a href="https://dom.spec.whatwg.org/#concept-element-local-name">local name</a> in <var>DEFINITION</var></li>
-        <li>Let <var>NAMESPACE</var> be the <a href="https://dom.spec.whatwg.org/#concept-element-namespace">namespace</a> in <var>DEFINITION</var></li>
-        <li>Set <var>ELEMENT</var>'s <a href="https://dom.spec.whatwg.org/#concept-element-local-name">local name</a> to <var>NAME</var>, <a href="https://dom.spec.whatwg.org/#concept-element-namespace">namespace</a> to the <var>NAMESPACE</var>, and <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a> to <var>DOCUMENT</var></li>
-        <li>If <var>TYPE</var> is not the same as <var>NAME</var>, set the value of <var>ELEMENT</var>'s <code>is</code> <a href="https://dom.spec.whatwg.org/#concept-named-attribute">attribute</a> to <var>TYPE</var></li>
-        <li><a data-lt="enqueue a lifecycle callback">Enqueue</a> <a data-lt="createdCallback">created</a> callback for <var>ELEMENT</var></li>
-        <li>Return <var>ELEMENT</var>.</li>
-    </ol></li>
+<section id="htmlelement-constructor">
+<h3>The <a href="https://html.spec.whatwg.org/multipage/dom.html#htmlelement"><code>HTMLElement</code></a> Constructor</h3>
+
+The <a href="https://html.spec.whatwg.org/multipage/dom.html#htmlelement"><code>HTMLElement</code></a> interface gains following annotation:
+
+<pre>
+[Constructor]
+</pre>
+
+The <dfn id="dom-htmlelement-constructor"><code>HTMLElement</code> constructor</dfn>, when invoked, must perform the following steps:
+
+<ol>
+    <li>Let <var>document</var> be the <a href="https://html.spec.whatwg.org/multipage/#concept-document-window">associated document</a> of <a href="https://html.spec.whatwg.org/#the-current-realm-record">the current Realm Record</a>'s [[\GlobalObject]].</li>
```

Comment:
As we discussed on IRC, I don't think this is quite right in the case a custom element extends another document's prototype (e.g. to have its own registry).  We should be using the global object of `[[Prototype]]` internal slot of the constructor or the global object of the realm of the `HTMLElement` constructor itself.